cli.go

```diff
@@ -163,16 +193,19 @@ func (cli *CLI) Run(args []string) int {
 				if err != nil {
 					return logError(err, ExitCodeRunnerError)
 				}
+				runner.SetReadyChannel(readyCh)
 				go runner.Start()
 			case *config.KillSignal:
 				fmt.Fprintf(cli.errStream, "Cleaning up...\n")
 				runner.StopImmediately()
+				cancel() // Cancel the context to stop the readyCh goroutine
```

Comment:
This is safe, but we also don't need it, because you have `defer cancel()` above and we're returning here.

```suggestion
```